### ISSUE TITLE
Question: why is the extension loading the .js files from https://stable.art even though they are bundled with it

### ISSUE THREAD

**Issue by nabileisa** [isekaidev/stable.art#12]
Hi, I was attempting to install the extension and kept failing to load with the error `Failed to load Stable.art`,
so I took a look at the code to try and figure out  what the problem was and turns out the scripts were being loaded from `https://stable.art/` not from the extension files .

```
<!doctype html><html><head><title>Stable.art</title><script defer="defer" onerror="handleScriptError()" src="https://stable.art/plugin-code/js/chunk-vendors.js"></script><script defer="defer" onerror="handleScriptError()" src="https://stable.art/plugin-code/js/app.js"></script></head><body><div id="app"><sp-heading size="L" style="text-align: center;" id="app-loader">Loading...</sp-heading></div><script>function handleScriptError() {
        document.getElementById('app-loader').textContent = 'Failed to load Stable.art';
      };</script></body></html>
```

**so,why is it loading the js files from a remote site instead of the files that are bundled with it?**

**Comment by isekaidev** [comment]
For updates - it allows users to get new versions almost in real time and I do not need to spend time creating separate ccx files for every update. Bundled files will be used in the future as a fallback for cases like yours, but I did not have time to implement it.

However, it is strange that you have this error. Could you open https://stable.art/plugin-code/js/app.js in the browser, does it work?

**Comment by nabileisa** [comment]
Understandable ,I just wish that mechanism and the `always online` requirement was mentioned in the `readme` for transparency at least.
as for the error I know the reason for it was because the machine I was testing it on was offline and photoshop had no access to the internet.
Thanks for the reply and the great work on this plugin and I look forward to future updates.

**Comment by ZeroCool22** [comment]
> Understandable ,I just wish that mechanism and the `always online` requirement was mentioned in the `readme` for transparency at least. as for the error I know the reason for it was because the machine I was testing it on was offline and photoshop had no access to the internet. Thanks for the reply and the great work on this plugin and I look forward to future updates.

This, would be good if the need of Online mode could be configurable by the user. So plugin works without connection.

**Comment by sstoryart1** [comment]
tried installing this plugin
installation went ok but all it says in big is loading...
does not work

**Comment by davidcovo** [comment]
![image](https://user-images.githubusercontent.com/23183521/229019942-20a61907-3e0d-4927-ad37-695941c0a1d3.png)

**Comment by yadyadayada** [comment]
New to Github, first comment ever, English is not my first language. I would love to add my voice here, if you can, would you please allow the plugin offline functionality?

**Comment by IWDLACSWCAKT** [comment]
Is there a similar app that DOESN'T need to be online to work?
That is a dealbreaker

**Comment by IWDLACSWCAKT** [comment]
Is there a workaround to the always online requirement? My internet is really inconsistent where I live right now and sometimes I like to use the plugin where there is no broadband

**Comment by GoregoriDes** [comment]
Go to the index.html and remove the http://localhost:8080
    <script defer onerror="handleScriptError()" src="http://localhost:8080/js/chunk-vendors.js"></script>  
    <script defer onerror="handleScriptError()" src="http://localhost:8080/js/app.js"></script>

Just keep it 
    <script defer onerror="handleScriptError()" src="/js/chunk-vendors.js"></script>  
    <script defer onerror="handleScriptError()" src="/js/app.js"></script>
and it will check in your local folder structure.

**Comment by IWDLACSWCAKT** [comment]
> Go to the index.html and remove the http://localhost:8080 <script defer onerror="handleScriptError()" src="http://localhost:8080/js/chunk-vendors.js"></script> <script defer onerror="handleScriptError()" src="http://localhost:8080/js/app.js"></script>
> 
> Just keep it <script defer onerror="handleScriptError()" src="/js/chunk-vendors.js"></script> <script defer onerror="handleScriptError()" src="/js/app.js"></script> and it will check in your local folder structure.

I can't find this on my index.html file
This is what I've got when I open it on visual studio code:

<!doctype html><html><head><title>Stable.art</title><script defer="defer" onerror="handleScriptError()" src="https://stable.art/plugin-code/js/chunk-vendors.js"></script><script defer="defer" onerror="handleScriptError()" src="https://stable.art/plugin-code/js/app.js"></script></head><body><div id="app"><sp-heading size="L" style="text-align: center;" id="app-loader">Loading...</sp-heading></div><script>function handleScriptError() {
        document.getElementById('app-loader').textContent = 'Failed to load Stable.art';
      };</script></body></html>

**Comment by IWDLACSWCAKT** [comment]
> Go to the index.html and remove the http://localhost:8080 <script defer onerror="handleScriptError()" src="http://localhost:8080/js/chunk-vendors.js"></script> <script defer onerror="handleScriptError()" src="http://localhost:8080/js/app.js"></script>
> 
> Just keep it <script defer onerror="handleScriptError()" src="/js/chunk-vendors.js"></script> <script defer onerror="handleScriptError()" src="/js/app.js"></script> and it will check in your local folder structure.

I did this and it worked, thank you!
But this generated other problems: 
-Generate button won't display % progress of current generation, it only displays how many seconds have passed (this is not an issue, although it might help someone understand what's going on better than me)
-Extra options no longer appear, meaning it will always create 4 different pictures, and I can't change the resolution when using inpaint.

**Comment by Shelushun** [comment]
Everything was working yesterday. Today, too, Failed to load stable.art.
I changed the file index.html but I lost the model selection settings and the image count settings. Is there a solution?

![Image](https://github.com/user-attachments/assets/f4ac8356-fd8e-4b12-bf85-9e605cc9e96c)

**Comment by Andreslav** [comment]
> Everything was working yesterday. Today, too, Failed to load stable.art. I changed the file index.html but I lost the model selection settings and the image count settings. Is there a solution?

I don't have that problem. Maybe it's because I have a newer version of the plugin.

Try replacing the local js in the files (/js/chunk-vendors.js, /js/app.js) to the one that is online (stable.art/plugin-code/...).

Don't forget to make a copy before making changes :)

**Comment by Shelushun** [comment]
> > Everything was working yesterday. Today, too, Failed to load stable.art. I changed the file index.html but I lost the model selection settings and the image count settings. Is there a solution?
> 
> I don't have that problem. Maybe it's because I have a newer version of the plugin.
> 
> Try replacing the local js in the files (/js/chunk-vendors.js, /js/app.js) to the one that is online (stable.art/plugin-code/...).
> 
> Don't forget to make a copy before making changes :)

Yes, the problem was that the servers became unavailable without vpn. Downloaded the files and replaced them in local folders. This is the only way that has helped. Thanks for the feedback!

**Comment by m0nox1** [comment]
@nabileisa @sstoryart1 @IWDLACSWCAKT @GoregoriDes @Shelushun @Andreslav has any of you solved this problem? could you help me solve the problem? after reading what you wrote here, I didn't understand what to do.

<img width="345" height="117" alt="Image" src="https://github.com/user-attachments/assets/7fe1c37e-a8b6-4f88-a671-33380aced1bd" />

**Comment by Andreslav** [comment]
@m0nox1, you need to edit the file index.html in the plugin folder: C:\Program Files\Adobe\Adobe Photoshop\Plug-ins\stable.art_0.0.1 (your path may vary).

Remove everything before "/js/" from the links. See the examples above.

**Comment by Shelushun** [comment]
> [@nabileisa](https://github.com/nabileisa) [@sstoryart1](https://github.com/sstoryart1) [@IWDLACSWCAKT](https://github.com/IWDLACSWCAKT) [@GoregoriDes](https://github.com/GoregoriDes) [@Shelushun](https://github.com/Shelushun) [@Andreslav](https://github.com/Andreslav) has any of you solved this problem? could you help me solve the problem? after reading what you wrote here, I didn't understand what to do.
> 
> <img alt="Image" width="345" height="117" src="https://private-user-images.githubusercontent.com/222484538/470511181-7fe1c37e-a8b6-4f88-a671-33380aced1bd.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NTM0MjI1ODUsIm5iZiI6MTc1MzQyMjI4NSwicGF0aCI6Ii8yMjI0ODQ1MzgvNDcwNTExMTgxLTdmZTFjMzdlLWE4YjYtNGY4OC1hNjcxLTMzMzgwYWNlZDFiZC5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjUwNzI1JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI1MDcyNVQwNTQ0NDVaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT0wMzQ0NWZhZDlkMzUwMmE2MDE4MjA0ZTk3Yzk2MjM2MGU2MjczYzk0ODZlYWRhMmVhZTVlMjRlYjU2NTJjMjUxJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.2Myl_Rst6O5IwPi-lW6VUYQP2l27HDFgJhlV2HDXRJ0">

https://disk.yandex.ru/d/eVmvpPH58nSPBQ
Here's my version. You can download and edit it. It doesn't depend on the paths

**Comment by Redskyinyoureyes** [comment]
@Shelushun God bless you man
